core/domain/exp_domain.py

```diff
@@ -2836,6 +2858,29 @@ def _convert_v33_dict_to_v34_dict(cls, exploration_dict):
 
         return exploration_dict
 
+    @classmethod
+    def _convert_v34_dict_to_v35_dict(cls, exploration_dict):
+        """Converts a v34 exploration dict into a v35 exploration dict.
+        Replaces tagged_misconception_id with tagged_skill_misconception_id,
+        which contains the skill id and misconception id of the tagged
+        misconception, connected by '-'.
+
+        Args:
+            exploration_dict: dict. The dict representation of an exploration
+                with schema version v34.
+
+        Returns:
+            dict. The dict representation of the Exploration domain object,
+            following schema version v35.
+        """
+        exploration_dict['schema_version'] = 35
+
+        exploration_dict['states'] = cls._convert_states_v29_dict_to_v30_dict(
+            exploration_dict['states'])
+        exploration_dict['states_schema_version'] = 30
+
+        return exploration_dict
+
     @classmethod
     def _migrate_to_latest_yaml_version(
```

Comment:
I think the 35 version needs to be tested (Point me to the test if I am missing it!)

Reply:
Ditto.